cmd/deploy/cfghandler.go

```diff
@@ -0,0 +1,91 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package deploy
+
+import (
+	"context"
+
+	"github.com/okteto/okteto/cmd/utils"
+	"github.com/okteto/okteto/pkg/cmd/pipeline"
+	"github.com/okteto/okteto/pkg/constants"
+	oktetoLog "github.com/okteto/okteto/pkg/log"
+	"github.com/okteto/okteto/pkg/okteto"
+	apiv1 "k8s.io/api/core/v1"
+)
+
+// configMapHandler defines the different functions to run okteto inside an okteto deploy
+// or an okteto destroy directly
+type configMapHandler interface {
+	translateConfigMapAndDeploy(context.Context, *pipeline.CfgData) (*apiv1.ConfigMap, error)
+	updateConfigMap(context.Context, *apiv1.ConfigMap, *pipeline.CfgData, error) error
+}
+
+// deployInsideDeployConfigMapHandler is the runner used when the okteto is executed
+// inside an okteto deploy command
+type deployInsideDeployConfigMapHandler struct{}
+
+func newDeployInsideDeployConfigMapHandler() *deployInsideDeployConfigMapHandler {
+	return &deployInsideDeployConfigMapHandler{}
+}
+
+// oktetoDefaultConfigMapHandler is the runner used when the okteto is executed
+// directly
+type defaultConfigMapHandler struct {
+	k8sClientProvider okteto.K8sClientProvider
+}
+
+func newDefaultConfigMapHandler(provider okteto.K8sClientProvider) *defaultConfigMapHandler {
+	return &defaultConfigMapHandler{
+		k8sClientProvider: provider,
+	}
+}
+
+func newConfigmapHandler(provider okteto.K8sClientProvider) configMapHandler {
+	if utils.LoadBoolean(constants.OKtetoDeployRemote) {
+		return newDeployInsideDeployConfigMapHandler()
+	}
+	return newDefaultConfigMapHandler(provider)
+}
+
+func (ch *defaultConfigMapHandler) translateConfigMapAndDeploy(ctx context.Context, data *pipeline.CfgData) (*apiv1.ConfigMap, error) {
+	c, _, err := ch.k8sClientProvider.Provide(okteto.Context().Cfg)
+	if err != nil {
+		return nil, err
+	}
+	return pipeline.TranslateConfigMapAndDeploy(ctx, data, c)
+}
+
+func (ch *defaultConfigMapHandler) updateConfigMap(ctx context.Context, cfg *apiv1.ConfigMap, data *pipeline.CfgData, errMain error) error {
+	c, _, err := ch.k8sClientProvider.Provide(okteto.Context().Cfg)
+	if err != nil {
+		return err
+	}
+	if errMain != nil {
+		oktetoLog.AddToBuffer(oktetoLog.ErrorLevel, errMain.Error())
+		data.Status = pipeline.ErrorStatus
+	}
+	if err := pipeline.UpdateConfigMap(ctx, cfg, data, c); err != nil {
+		return err
+	}
+	return errMain
+}
+
+func (*deployInsideDeployConfigMapHandler) translateConfigMapAndDeploy(_ context.Context, _ *pipeline.CfgData) (*apiv1.ConfigMap, error) {
+	return nil, nil
+}
+
+func (*deployInsideDeployConfigMapHandler) updateConfigMap(_ context.Context, _ *apiv1.ConfigMap, _ *pipeline.CfgData, err error) error {
+	oktetoLog.AddToBuffer(oktetoLog.InfoLevel, "Destruction failed: %s", err.Error())
+	return nil
```

Comment:
why the methods have no implementation?

Reply:
Because we don't have to update the configmap if it's executed in the remote deploy

Reply:
can we add an explanatory comment about why?

Reply:
Will do!!

Reply:
Wouldn't it be better to check if we are in remote or not and do the update if we are not? instead of leaving it to the interface? I say this because in this case having an empty implementation seems strange to me

Reply:
Having an if steatement around in every part of the code that calls these functions is also a smell for me. Having an interface makes the tests more unitary since the code does not check an environment variable that deprives us of launching the tests in parallel